exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -347,6 +347,59 @@ func (ku *Kucoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) err
 	return nil
 }
 
+// GetFuturesTickers does n * REST requests based on enabled pairs of the futures asset type
+func (ku *Kucoin) GetFuturesTickers(ctx context.Context) ([]*ticker.Price, error) {
+	pairs, err := ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		return nil, err
+	}
+
+	var wg sync.WaitGroup
+	tickersC := make(chan *ticker.Price, len(pairs))
+	errC := make(chan error, len(pairs))
+
+	for i := range pairs {
+		p, err := ku.FormatExchangeCurrency(pairs[i], asset.Futures)
+		if err != nil {
+			errC <- err
+			break
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			if tick, err := ku.GetFuturesTicker(ctx, p.String()); err != nil {
+				errC <- err
+			} else {
+				tickersC <- &ticker.Price{
+					Last:         tick.Price.Float64(),
+					Bid:          tick.BestBidPrice.Float64(),
+					Ask:          tick.BestAskPrice.Float64(),
+					Volume:       tick.Size,
+					Pair:         p,
+					LastUpdated:  tick.FilledTime.Time(),
+					ExchangeName: ku.Name,
+					AssetType:    asset.Futures,
+				}
+			}
+		}()
+	}
+
+	wg.Wait()
+	close(tickersC)
+	close(errC)
+
+	for err := range errC {
```

Comment:
Looks like the intent was to capture all errors for all pairs, but we're returning on the first occurrence of an error. Did you mean to append a bunch of errors here?

Reply:
added an append

Reply:
Nice! But the append should be used in the for loop. This should do it:

```
	var errs error
	for err := range errC {
		errs = common.AppendError(errs, err)
	}

	if errs != nil {
		return nil, errs
	}
```